### PR TITLE
Issue 9113 refactor g vehicle sound params list

### PR DESCRIPTION
**Pull request by UltimaBGD**
This is a refactor of the gVehicleSoundParams List as referenced in #9113.

I have renamed it to vehicleSoundParams and initialized it within the vehicle_sound_update, and pass it to vehicle_update_sound_params.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
Did you actually manage to compile it yourself? All CIs are red.

**Comment by UltimaBGD** [comment]
My apologies, it seems I am having some issues on my end with Visual Studio. I will get back on this very shortly.

File: .vscode/c_cpp_properties.json [hunk 1]
```diff
@@ -90,4 +90,4 @@
         }
     ],
     "version": 4
-}
```
Comment (Gymnasiast) [inline .vscode/c_cpp_properties.json]: Could you revert this change?

Comment (UltimaBGD) [reply .vscode/c_cpp_properties.json]: Should be resolved now. The github editor seemed to have been auto adding the newline, so sme of my commits have no changes because of it.

**Comment by UltimaBGD** [comment]
The only changes were to that one file, so I am unsure why Travis is erroring on certain builds now.

**Comment by duncanspumpkin** [comment]
Hmm this certainly works but I'm thinking std::array is horrible for this. std::vector isn't great either due to the memory allocations.

**Comment by UltimaBGD** [comment]
std::array was specified in the initial issue, so that is why I ended up using it.

**Comment by duncanspumpkin** [comment]
Yeah i know. Just having a think of anyway to improve it.

**Comment by Gymnasiast** [comment]
Continued as https://github.com/OpenRCT2/OpenRCT2/pull/10596.
